src/plugins/htmlVideoPlayer/plugin.js

```diff
@@ -448,11 +470,36 @@ function tryRemoveElement(elem) {
             destroyFlvPlayer(this);
             destroyCastPlayer(this);
 
+            let secondaryTrackValid = true;
+
             this.#subtitleTrackIndexToSetOnPlaying = options.mediaSource.DefaultSubtitleStreamIndex == null ? -1 : options.mediaSource.DefaultSubtitleStreamIndex;
             if (this.#subtitleTrackIndexToSetOnPlaying != null && this.#subtitleTrackIndexToSetOnPlaying >= 0) {
                 const initialSubtitleStream = options.mediaSource.MediaStreams[this.#subtitleTrackIndexToSetOnPlaying];
                 if (!initialSubtitleStream || initialSubtitleStream.DeliveryMethod === 'Encode') {
                     this.#subtitleTrackIndexToSetOnPlaying = -1;
+                    secondaryTrackValid = false;
+                }
+                // secondary track should not be shown if primary track is no longer a valid pair
+                if (initialSubtitleStream && !playbackManager.trackHasSecondarySubtitleSupport(initialSubtitleStream)) {
+                    secondaryTrackValid = false;
+                }
+            } else {
+                secondaryTrackValid = false;
+            }
+
+            // Get the secondary track that has been set during this watch session
+            let currentSecondaryTrackIndex = playbackManager.getSecondarySubtitleStreamIndex();
+
+            if (!secondaryTrackValid) {
+                currentSecondaryTrackIndex = -1;
+                playbackManager.setSecondarySubtitleStreamIndex(currentSecondaryTrackIndex);
```

Comment:
This should go after
https://github.com/jellyfin/jellyfin-web/blob/f4abbf3e22aec8c3743e819d3764890aba98277a/src/plugins/htmlVideoPlayer/plugin.js#L508

Strange that this issue didn't occur with `devtool: 'source-map'` in `webpack.dev.js`.

Reply:
Oh hmm. Okay let me take a look at this

Reply:
Alright, the most recent 2 commits fix these issues. One for the bug, and the other for clearing and setting the index.

There are soooo many different states being passed and saved, it is really difficult to keep track of what is doing what. I tried to set the minimum amount to allow the secondary subtitles to be carried over to the next media source when playing multiple episodes of a show.

I also decided to not keep the saved secondary track index state once you back out. As you mentioned, it was getting saved to the player state so it would continue to exist for the current session while it is loaded.
But after thinking about this it doesn't make sense to keep the index stored like that. The user can go from an episode to a movie. They might be entirely unrelated with subtitles, so the index is not useful in that case.

Since we do not have any stored secondary subtitle language settings that we can pull from, similar to how subtitle and audio languages are fetched, I don't think it is appropriate to include at the moment.

So now, the flow for auto playing secondary subtitles is:
```
user clicks on a video -> 
selects the secondary subtitle -> 
clicks to next/prev video from the player -> 
the currently active secondary subtitle goes into the current `rankStreamType` logic to find the best match
```
https://github.com/jellyfin/jellyfin-web/blob/f4abbf3e22aec8c3743e819d3764890aba98277a/src/components/playback/playbackmanager.js#L2314